### ISSUE TITLE
Release 5.0.0b0

### ISSUE THREAD

**Issue by ashwin153** [mkorpela/overrides#71]
- Would it be possible to release a beta version of `0.5.0` to PyPI? I have a fairly large class hierarchy in my project that would exercise a lot of the new functionality (e.g., everything is type-hinted, there are self-types, there are static methods, etc.). Unfortunately, changing the build infrastructure to use git references instead of pip packages is a little hard in this project. Getting `0.5.0` to work on my project should give us additional confidence going into the `0.5.0` release.

**Comment by mkorpela** [comment]
I think next would be 5.0.0 but sure a beta (preversion) could be done.

**Comment by mkorpela** [comment]
I would still like a solution to https://github.com/mkorpela/overrides/issues/61

**Comment by ashwin153** [comment]
I still think that it is an invalid override that should be fixed or marked with `ignore_signature`; the signature of the subclass method is not compatible with the superclass method.

**Comment by mkorpela** [comment]
I think the signature outside of the method is correctly overriding. Internal state is different when values drop to a bit different positions depending on how it is called.

**Comment by ashwin153** [comment]
In my opinion, behavior should be the same when using positional and keyword parameters. In other words, if I have a method `def foo(x)` I would expect that `foo(1)` and `foo(x=1)` should produce the same result. Fundamentally, I believe that the choice between passing `x` via a positional parameter or a keyword one is a choice of aesthetics and not of behavior. However, in the linked issue positional and keyword calls to the same method have completely different results.

**Comment by brentyi** [comment]
If anybody's curious what mypy does...

Checking:
```python
class A:
    def methoda(self, x: int = 0) -> None:
        pass


class B(A):
    def methoda(self, y: int = 0, **kwargs) -> None:
        pass
```

Results in:
```
test.py:7: error: Signature of "methoda" incompatible with supertype "A"
Found 1 error in 1 file (checked 1 source file)
```

**Comment by mkorpela** [comment]
Current HEAD of master does now think that doing that is just fine.

To me the signature is the public border.

Whatever changes happen in the public side of that should be only in the consideration.
This is in my opinion also currently the safer approach.
We are now making the assertions very much stricter than before.

If someone can come up with an example where a valid super class method call would be invalid when splitting checks to positional and keyword call then I would be willing to change my opinion here. Or if now emerges a relevant failure pattern that is caused by this.

There is always possibility to make them even stricter later. Beta sounds great and going to do that now.

**Comment by mkorpela** [comment]
Pre-Release is out https://pypi.org/project/overrides/5.0.0b0/
- I'll close this issue now => We can start a new one if needed about signature verification differences with mypy.

**Comment by brentyi** [comment]
Thanks @mkorpela !

**Comment by mkorpela** [comment]
@ashwin153 I've done some changes related to odd signatures. Still I'm thinking that the solution is not bulletproof but might be close enough for now.

If your check with your project do not reveal problems, I think a release could be in place :)
I would like to thank @ashwin153, @brentyi and @cboots - you have all made this happen!

If it is ok, I would like to announce this version so that you all are mentioned as contributors. Especially @ashwin153 - thanks for huge contributions!

**Comment by brentyi** [comment]
Sounds great.

I just checked against the latest version of `overrides`: seems like things are pretty much at the point where my project could work with the latest updates.

The biggest pain point for me is forward references. I listed some examples in #63, but with the current setup it seems like I'll need to go function-by-function throughout my projects and selectively set `check_signature` to `False` (for cases that use `if TYPE_CHECKING`) or `check_at_runtime` to `True` (for general forward references) based on context, and then keep these updated under any code refactoring. I have downstream users who are subclassing from what I'm writing, and I'd have to ask them to do the same.

Another thing that's causing errors (which is valid but likely impossible to support) is class methods overriding static methods and vice versa:
```
import abc
from typing import Type, TypeVar

from overrides import EnforceOverrides, overrides

T = TypeVar("T")


class A(EnforceOverrides):
    @classmethod
    @abc.abstractmethod
    def make(cls: Type[T]) -> T:
        ...


class B(A):
    @staticmethod
    @overrides(check_at_runtime=True)
    def make() -> "B":
        return B()


B.make(0)
```

As a thought, particularly if these things can't be resolved: would it be possible to introduce something like a `EnforceOverridesNoSignatureChecks`/`EnforceOverridesLegacy` base class? Would make migrating old code easier.
